x-pack/plugins/security/public/analytics/register_user_context.ts

```diff
@@ -34,22 +34,28 @@ export function registerUserContext(
     context$: from(authc.getCurrentUser()).pipe(
       map((user) => {
         if (user.elastic_cloud_user) {
+          let userId = user.username;
           // If the user is managed by ESS, use the plain username as the user ID:
           // The username is expected to be unique for these users,
           // and it matches how users are identified in the Cloud UI, so it allows us to correlate them.
-          return { userId: user.username, isElasticCloudUser: true };
+
+          if (!/^[0-9]+$/.test(userId)) {
+            // If the userId is not a number, we don't want to risk reporting any PII, so we choose to hash it.
+            userId = sha256(userId);
+          }
```

Comment:
As an extra precaution, we are checking that the userId is a numeric value. If not, we hash it.

Reply:
Why hash non-numerical username but not numerical usernames? The username could be a telephone number which would also considered to be PII. I don't think there's a reliable regex to determine whether something is PII.